### PR TITLE
Bmi unit test

### LINKED ISSUE #13: reimplement `get_var_grid` def
Definition for `Get_var_grid()` in [bmi_cfe.c](https://github.com/NOAA-OWP/cfe/blob/965109f764166d25cab6dbae2b6740d85bf279a1/src/bmi_cfe.c#L907) is commented-out.  Assignment error has been fixed so need to just reinstate def

All other functions pass unit test.

### PR DESCRIPTION
**Pull request by madMatchstick**
Include BMI unit testing for CFE
- unit test C script [main_unit_test_bmi.c](https://github.com/madMatchstick/cfe/blob/bmi-unit-test/test/main_unit_test_bmi.c)
- make and run shell [make_and_run_bmi_unit_test.sh](https://github.com/madMatchstick/cfe/blob/bmi-unit-test/test/make_and_run_bmi_unit_test.sh)
- configurations & forcings
- [README](https://github.com/madMatchstick/cfe/blob/bmi-unit-test/test/README.md) gives instructions on how to run 
- Redefining `get_var_grid()` def. Fixes #13 

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: test/make_and_run_bmi_unit_test.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+gcc -lm main_unit_test_bmi.c ../src/bmi_cfe.c ../src/cfe.c -o run_cfe_bmi_test
+./run_cfe_bmi_test ./configs/cat_89_bmi_config_cfe.txt
```
Comment (jmframe) [inline test/make_and_run_bmi_unit_test.sh:3]: In continuation of the comment above, we should be able to run pointing to a config file in the other configs directory. We can even have a config file for the unit test: 
`./run_cfe_bmi_test ../configs/cat_89_bmi_config_cfe_unit_test.txt`  

I don't know what would be better, just bringing it up for discussion.

File: test/configs/cat_89_bmi_config_cfe.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+forcing_file=./forcings/cat89_01Dec2015.csv
```
[the remark this answers is not in the document]
Comment (madMatchstick) [reply test/configs/cat_89_bmi_config_cfe.txt:1]: yes, we can. I'll adjust this.

File: test/make_and_run_bmi_unit_test.sh (hunk 1 above)
Comment (madMatchstick) [reply test/make_and_run_bmi_unit_test.sh:3]: will update.  good suggestion.

**Review by jmframe — APPROVED** [review]
nice

**Pull request merged**